code/controllers/subsystems/news.dm

```diff
@@ -0,0 +1,169 @@
+	var/datum/feed_message/wanted_issue
+
+/datum/controller/subsystem/news/Recover()
+	src.network_channels = SSnews.network_channels
+	src.wanted_issue = SSnews.wanted_issue
+
+/datum/controller/subsystem/news/New()
+	NEW_SS_GLOBAL(SSnews)
+
+/datum/controller/subsystem/news/Initialize(timeofday)
+	CreateFeedChannel("Station Announcements", "Automatic Announcement System", 1, 1, "New Station Announcement Available")
+	CreateFeedChannel("Tau Ceti Daily", "CentComm Minister of Information", 1, 1)
+	CreateFeedChannel("The Gibson Gazette", "Editor Carl Ritz", 1, 1)
+	if(config.sql_enabled)
+		load_from_sql()
+	..()
+
+/datum/controller/subsystem/news/proc/load_from_sql()
+	if(!establish_db_connection(dbcon))
+		log_debug("SSnews: SQL ERROR - Failed to connect.")
+		return
+	var/DBQuery/channel_query = dbcon.NewQuery("SELECT id, name, author, locked, is_admin_channel, announcement FROM ss13_news_channels WHERE deleted_at IS NULL ORDER BY name ASC")
+	channel_query.Execute()
+	while(channel_query.NextRow())
+		CHECK_TICK
+		var/datum/feed_channel/channel = null
+		try
+			channel = CreateFeedChannel(
+				channel_query.item[2],
+				channel_query.item[3],
+				text2num(channel_query.item[4]),
+				text2num(channel_query.item[5]),
+				channel_query.item[6])
+		catch(var/exception/ec)
+			log_debug("SSnews: Error when loading channel: [ec]")
+			continue
+		var/DBQuery/news_query = dbcon.NewQuery("SELECT body, author, is_admin_message, message_type, time_stamp FROM ss13_news_stories WHERE deleted_at IS NULL AND channel_id = :channel_id: ORDER BY created_at DESC")
+		news_query.Execute(list("channel_id" = channel_query.item[1]))
+		while(news_query.NextRow())
+			CHECK_TICK
+			try
+				SubmitArticle(news_query.item[1], news_query.item[2], channel, null, text2num(news_query.item[3]), news_query.item[4], news_query.item[5])
+			catch(var/exception/en)
+				log_debug("SSnews: Error when loading news: [en]")
+
+/datum/controller/subsystem/news/proc/GetFeedChannel(var/channel_name)
+//	for(var/datum/feed_channel/FC in network_channels)
+//		if(FC.channel_name == channel_name)
+//			return FC
+//	return null
+	if(network_channels[channel_name])
+		return network_channels[channel_name]
+	return null
+
+/datum/controller/subsystem/news/proc/CreateFeedChannel(var/channel_name, var/author, var/locked, var/adminChannel = 0, var/announcement_message)
+	var/datum/feed_channel/newChannel = new /datum/feed_channel
+	newChannel.channel_name = channel_name
+	newChannel.author = author
+	newChannel.locked = locked
+	newChannel.is_admin_channel = adminChannel
+	if(announcement_message)
+		newChannel.announcement = announcement_message
+	else
+		newChannel.announcement = "Breaking news from [channel_name]!"
+	network_channels[channel_name] = newChannel
+	return newChannel
+
+/datum/controller/subsystem/news/proc/SubmitArticle(var/msg, var/author, var/channel, var/obj/item/weapon/photo/photo, var/adminMessage = 0, var/message_type = "", var/time_stamp)
```

Comment:
For the sake of clarity, please declare `var/channel` as `var/datum/feed_channel/channel`. DM does implicit type-casting, which means you can still send text as the argument there. The code as you wrote it will work just fine with this change, but it makes the proc more readable.

This also allows you to change the check on line 75 to just `if (!istype(channel))`